### PR TITLE
Allow this.function.selector as bytesN and in pure mutability

### PR DESCRIPTION
**Pull request by seanyoung**
Improve compatibility with solc.

### CODE REVIEW

File: src/sema/mutability.rs [hunk 1]
```diff
@@ -252,6 +252,18 @@ fn read_expression(expr: &Expression, state: &mut StateCheck) -> bool {
         Expression::Subscript { loc, array_ty, .. } if array_ty.is_contract_storage() => {
             state.read(loc)
         }
+        Expression::Builtin {
+            kind: Builtin::FunctionSelector,
```
Comment (xermicus) [inline src/sema/mutability.rs:256]: Is this binding necessary?

**Review by xermicus — APPROVED** [review]

File: src/sema/mutability.rs (hunk 1 above)
Comment (seanyoung) [reply src/sema/mutability.rs:256]: I am not following.  Here we are filtering on `kind == Builtin::FunctionSelector`

File: tests/contract_testcases/solana/selector.sol [hunk 2]
```diff
@@ -0,0 +1,7 @@
+contract c {
+    function g() pure public returns (bytes8) {
+        return this.f.selector ^ this.x.selector;
+    }
```
Comment (LucasSte) [inline tests/contract_testcases/solana/selector.sol:4]: This deserves a runtime test.

**Review by LucasSte — COMMENTED** [review]
Why is this change necessary?

File: src/sema/mutability.rs (hunk 1 above)
Comment (xermicus) [reply src/sema/mutability.rs:256]: Of course, I was not paying attention

File: tests/contract_testcases/solana/selector.sol (hunk 2 above)
Comment (seanyoung) [reply tests/contract_testcases/solana/selector.sol]: Runtime test added

**Pull request merged**